docs/source/daml/intro/10_Functional101.rst

```diff
@@ -116,29 +116,36 @@ If we want to partially apply an infix operation we can also do that as follows:
 
 Type Constraints
 ................
 
-The ``Additive a =>`` part of the signature of ``add`` is a type constraint on the type parameter ``a``. ``Additive`` here is a typeclass. You already met typeclasses like ``Eq`` and ``Show`` in :doc:`3_Data`. The ``Additive`` typeclass says that you can add a thing. Ie there is a function ``(+) : a -> a -> a``. Now the way to read the full signature of ``add`` is "Given that a has an instance for the Additive typeclass, a maps to a function which maps a to a".
+The ``Additive a =>`` part of the signature of ``add`` is a type constraint on the type parameter ``a``. ``Additive`` here is a typeclass. You already met typeclasses like ``Eq`` and ``Show`` in :doc:`3_Data`. The ``Additive`` typeclass says that you can add a thing, i.e. there is a function ``(+) : a -> a -> a``. Now the way to read the full signature of ``add`` is "Given that ``a`` has an instance for the ``Additive`` typeclass, ``a`` maps to a function which maps ``a`` to ``a``".
 
-Typeclasses in Daml are a bit like interfaces in other languages. To be able to add two things using the ``+`` function, those things need to expose the ``+`` interface.
+Typeclasses in Daml are a bit like interfaces in other languages. To be able to add two things using the ``+`` function, those things need to "expose" (have an instance for) the ``Additive`` interface (typeclass).
 
 Unlike interfaces, typeclasses can have multiple type parameters. A good example, which also demonstrates the use of multiple constraints at the same time, is the signature of the ``exercise`` function:
 
 .. code-block:: daml
 
   exercise : (Template t, Choice t c r) => ContractId t -> c -> Update r
 
-Let's turn this into prose: Given that ``t`` is the type of a template, and that ``t`` has a choice ``c`` with return type ``r``, map a ``ContractId`` for a contract of type ``t`` to a function that takes the choice arguments of type ``c`` and returns an ``Update`` resulting in type ``r``.
+Let's turn this into prose: Given that ``t`` is the type of a template, and that ``t`` has a choice ``c`` with return type ``r``, the ``exercise`` function maps a ``ContractId`` for a contract of type ``t`` to a function that takes the choice arguments of type ``c`` and returns an ``Update`` resulting in type ``r``.
 
 That's quite a mouthful, and does require one to know what *meaning* the typeclass ``Choice`` gives to parameters ``t`` ``c`` and ``r``, but in many cases, that's obvious from the context or names of typeclasses and variables.
 
+Note that using single letters, while common, is not mandatory. The above may be made a little bit clearer by expanding the type parameter names, at the cost of making the code a bit longer:
+
+.. code-block:: daml
+
+  exercise : (Template template, Choice template choice result) =>
+               ContractId template -> choice -> Update result
```

Comment:
I like this a lot. Single letters and unreadable shortenings are terrible.

LGTM.

:trollface: